bugsnag-plugin-android-exitinfo/src/main/java/com/bugsnag/android/ExitInfoCallback.kt

```diff
@@ -0,0 +1,30 @@
+package com.bugsnag.android
+
+import android.app.ActivityManager
+import android.app.ApplicationExitInfo
+import android.content.Context
+import android.os.Build
+import androidx.annotation.RequiresApi
+
+internal class ExitInfoCallback(
+    private val context: Context,
+    private val eventEnhancer: (Event, ApplicationExitInfo) -> Unit
+) : OnSendCallback {
+
+    @RequiresApi(Build.VERSION_CODES.R)
+    override fun onSend(event: Event): Boolean {
+        val am = context.getSystemService(Context.ACTIVITY_SERVICE) as ActivityManager
+        val allExitInfo = am.getHistoricalProcessExitReasons(context.packageName, 0, MAXIMUM_NUMBER)
+        allExitInfo.forEach { exitInfo ->
+            if (event.session?.id?.let { exitInfo.processStateSummary?.contentEquals(it.toByteArray()) } == true) {
+                eventEnhancer(event, exitInfo)
+                return@onSend true
+            }
+        }
+        return true
+    }
+
+    internal companion object {
+        const val MAXIMUM_NUMBER = 100
```

Comment:
A more descriptive name would be good:

```suggestion
        const val MAX_EXIT_INFO = 100
```